ng2-components/ng2-activiti-processlist/src/components/activiti-start-process.component.spec.ts

```diff
@@ -80,79 +72,217 @@ describe('ActivitiStartProcessButton', () => {
+                expect(optionEl.textContent.trim()).toBe('My Process 1');
+                done();
+            });
+        });
+
     });
 
-    it('should call service to start process with the correct parameters', (done) => {
-        component.name = 'My new process';
-        component.showDialog();
-        fixture.detectChanges();
-        component.onChange('my:process1');
-        component.startProcess();
-        fixture.whenStable().then(() => {
-            expect(startProcessSpy).toHaveBeenCalledWith('my:process1', 'My new process', undefined);
-            done();
+    describe('input changes', () => {
+
+        let change = new SimpleChange('123', '456');
+        let nullChange = new SimpleChange('123', null);
+
+        beforeEach(async(() => {
+            component.appId = '123';
+            fixture.detectChanges();
+            fixture.whenStable().then(() => {
+                fixture.detectChanges();
+                getDefinitionsSpy.calls.reset();
+            });
+        }));
+
+        it('should reload processes when appId input changed', () => {
+            component.ngOnChanges({ appId: change });
+            expect(getDefinitionsSpy).toHaveBeenCalledWith('456');
         });
+
+        it('should reload processes when appId input changed to null', () => {
+            component.ngOnChanges({ appId: nullChange });
+            expect(getDefinitionsSpy).toHaveBeenCalledWith(null);
+        });
+
+        it('should not reload processes when changes do not include appId input', () => {
+            component.ngOnChanges({});
+            expect(getDefinitionsSpy).not.toHaveBeenCalled();
+        });
+
     });
 
-    it('should output start event when process started successfully', (done) => {
-        let emitSpy = spyOn(component.start, 'emit');
-        component.name = 'My new process';
-        component.showDialog();
-        fixture.detectChanges();
-        component.onChange('my:process1');
-        component.startProcess();
-        fixture.whenStable().then(() => {
-            expect(emitSpy).toHaveBeenCalledWith(newProcess);
-            done();
+    describe('start process', () => {
+
+        beforeEach(() => {
+            component.name = 'My new process';
+            component.showDialog();
+            fixture.detectChanges();
         });
+
+        it('should call service to start process if required fields provided', async(() => {
+            component.onProcessDefChange('my:process1');
+            component.startProcess();
+            fixture.whenStable().then(() => {
+                expect(startProcessSpy).toHaveBeenCalled();
+            });
+        }));
+
+        it('should avoid calling service to start process if required fields NOT provided', async(() => {
+            component.name = '';
+            component.startProcess();
+            fixture.whenStable().then(() => {
+                expect(startProcessSpy).not.toHaveBeenCalled();
```

Comment:
Added more tests to check the state of the start button (disabled or not) in 3ef10db9